plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -1279,6 +1320,121 @@ public static Optional<ReaderColumns> projectColumns(List<IcebergColumnHandle> c
         return Optional.of(new ReaderColumns(projectedColumns.build(), outputColumnMapping.build()));
     }
 
+    /**
+     * Creates a set of sufficient columns for the input projected columns and prepares a mapping between the two.
+     * For example, if input {@param columns} include columns "a.b" and "a.b.c", then they will be projected
+     * from a single column "a.b".
```

Comment:
Do we have a test that hits this case?

Reply:
Added `TestIcebergParquetConnectorTest.testProjectionPushdownReadsOnlyTheEnclosingColumn`